homeassistant/components/tellstick.py

```diff
@@ -0,0 +1,216 @@
+"""
+Tellstick Component.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/Tellstick/
+"""
+import logging
+import threading
+
+from homeassistant import bootstrap
+from homeassistant.const import (
+    ATTR_DISCOVERED, ATTR_SERVICE,
+    EVENT_PLATFORM_DISCOVERED, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.loader import get_component
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = "tellstick"
+
+REQUIREMENTS = ['tellcore-py==1.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_SIGNAL_REPETITIONS = "signal_repetitions"
+
+SIGNAL_REPETITIONS = 1
+
+DISCOVER_SWITCHES = "tellstick.switches"
+DISCOVER_LIGHTS = "tellstick.lights"
+DISCOVERY_TYPES = {"switch": DISCOVER_SWITCHES,
+                   "light": DISCOVER_LIGHTS}
+
+DISCOVER_DEVICES = "devices"
+DISCOVER_CONFIG = "config"
+
+# Use a global tellstick domain lock to handle
+# tellcore errors then calling to concurrently
+TELLSTICK_LOCK = threading.Lock()
```

Comment:
No, It's not used in the registry, it's used in the `TellstickDevice` base class. 

I could add it as an static class var instead if you think that's a better solution. It can't be initialized in the `__init__` method of the base class because it needs to be global for all device instances. Doing it in the Registry __init__ when it's not used there seems wrong.